src/core/crash-handler.c

```diff
@@ -21,13 +22,21 @@ _noreturn_ void freeze_or_exit_or_reboot(void) {
         /* If we are running in a container, let's prefer exiting, after all we can propagate an exit code to
          * the container manager, and thus inform it that something went wrong. */
         if (detect_container() > 0) {
+                (void) sd_notify(/* unset_environment= */ false, "EXIT_STATUS=1\n");
```

Comment:
This happens in the signal handler, is it safe to do these calls? We can't allocate memory, etc

Reply:
I didn't check too thoroughly but `sd_notify()` doesn't seem to allocate on a superficial look.

Reply:
It does call getenv which is not listed as signal safe in the manpage as far as I can tell

Reply:
Dropped the notify stuff